pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -28,38 +29,48 @@
 
   metalSupport ? stdenv.isDarwin && stdenv.isAarch64,
 
+  # You can find a full list here: https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
+  # For example if you're on an GTX 1080 that means you're using "Pascal" and you need to pass "sm_60"
+  arches ? cudaPackages.cudaFlags.arches or [ ],
+
   # You can find list of x86_64 options here: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
   # For ARM here: https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html
   # If you set "march" to "native", specify "mtune" as well; otherwise, it will be set to "generic". (credit to: https://lemire.me/blog/2018/07/25/it-is-more-complicated-than-i-thought-mtune-march-in-gcc/)
-  # For example, if you have an AMD Ryzen CPU, you will set "march" to "x86-64" and "mtune" to "znver2"
   march ? "",
+  # Apple Silicon Chips (M1, M2, M3 and so on DO NOT use mtune)
+  # For example, if you have an AMD Ryzen CPU, you will set "mtune" to "znver2"
   mtune ? "",
+  mcpu ? "",
```

Comment:
I'm still not convinced it's worth introducing the callPackage arguments. Conceptually, it'd be more consistent to require users to use something like:

```nix
let
  withMarch = march: oldAttrs: {
    env = oldAttrs.env // { NIX_CFLAGS_COMPILE = env.NIX_CFLAGS_COMPILE or "" + "-march=${march}"; };
  };
  pkgs' = mapAttrs (_: withMarch "x86_64-v3") pkgs;
in {
  inherit (pkgs') koboldcpp llama-cpp ollama;
}
```

It's a bit of a mouthful, but it's consistent and (fairly) universal. To contrast, while you could `koboldcpp.override { march = "..."; }`, you couldn't `blender.override { march = "..."; }` because it's an ad hoc argument.

I guess it wouldn't hurt much to let this through, I'll look at this again in the morning. I'll try not to delay merging any further than that

Reply:
That's a good point, thanks for bringing this up. I guess it's more appropriate to let users "merge/append" `march`, `mtune`, and `mcpu` to `env` themselves. I was mainly introducing these options because they were easier from a users POV, but since other AI packages already have this convention, it would be nice on my part to follow it too.